contents/data/databases.md

```diff
@@ -0,0 +1,115 @@
+| Fruit | A fruit.                     | 1.00  |      100      |
+| Bread | Sliced for your convenience. | 1.40  |       50      |
+| Water | Essential for life.          | 0.50  |     1000      |
+
+From this visualisation, we can define the different parts of the relation:
+
+- The whole table is known as a **relation**.
+- Each data row (i.e. excluding the header row) is known as a **tuple**, for instance: (Fruit, A fruit., 1.00, 100).
+- Each column is an **attribute** and each attribute has a **domain** or **data type**. The *Current Stock* column of the table is an attribute with a data type of integer (as product stock is logically represented using an integer).
+- Each element in a tuple is called an **attribute value**
+
+[Click here](https://en.wikipedia.org/wiki/Relation_(database)) for a more in-depth and formal definition of relations.
+
+## Transactions
+
+Databases are useless if the data cannot be used in a meaningful manner. However, uncontrolled access to the database would not be ideal, as it might cause problems, especially when other actions may depend on the previous action's result.
+
+Take for instance, a program that transfers money from one bank account to another. The actions that the program would need to do involves:
+
+1. Checking if the sender has enough money
+2. Deducting the amount of money to be sent
+3. Adding that amount to the receiver's account
+
+A problem arises if another action takes place in between any of the steps (e.g. another transfer from the same sender), or if any of the steps fails (e.g. due to a program crash). As a result, it may cause the data to be manipulated in an undesired manner (e.g. money not credited to receiver).
+
+Thus, **transactions** allow us to guard against these problems. A **transaction** symbolises a logical unit of work, which consists of multiple database actions, performed on a set of databases. [[Source]](https://en.wikipedia.org/wiki/Database_transaction) Properties of a database transaction ensures that these actions are done in a predictable and reliable manner.
+
+Thus, the transaction that would be implemented in the program could be something like this:
+
+1. Perform funds transfer from *sender* to *receiver*
+    - Checking if the sender has enough money
+    - Deducting the amount of money to be sent
+    - Adding that amount to the receiver's account
+
+As demonstrated above, the actions that needs to be done in order to transfer funds is wrapped as one large **transaction**. Hence, the *funds transfer* can be seen as the unit of work to be done on the database. The actions that make up the **transaction** are executed as though **transaction** is a single action.
+
+### ACID in transactions
+
+For a **transaction** to be considered as an implementation of the **transaction** concept, it has to satisfy the ACID principle.
+
+This principle states that a **transaction** must contain these characteristics:
```

Comment:
we usually do not describe for a transaction to have ACID characteristics. rather, RDBMS follows ACID properties because we want to ensure that data is stored predictability according to the statements in the transactions.